packages/ckeditor5-dev-build-tools/tests/build/arguments.test.ts

```diff
@@ -0,0 +1,213 @@
+/**
+ * @license Copyright (c) 2003-2024, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+import { test, expect, vi } from 'vitest';
+import fs from 'fs';
+import * as config from '../../src/config.js';
+import { build } from '../../src/build.js';
+
+/**
+ * Mock `rollup`, so it doesn't try to build anything.
+ */
+vi.mock( 'rollup', () => ( {
+	rollup() {
+		return {
+			write: () => {}
+		};
+	}
+} ) );
+
+/**
+ * Mock function for generating rollup configuration.
+ */
+const spy = vi
+	.spyOn( config, 'getRollupConfig' )
+	.mockImplementation( (): any => {} );
```

Comment:
I noticed that none of the mocks are cleared or reset after being used in a test. In multiple tests it is checked, for example, whether a mock was called with specific arguments, so it may happen that a given test passes by a chance, because some previous mock call matches current expectations.

I think the easiest way to handle it is by configuring `vitest.config.ts` by adding [`restoreMocks`](https://vitest.dev/config/#restoremocks) flag.